pages/docs/reference/java-interop.md

```diff
@@ -131,6 +131,135 @@ Kotlin types. The compiler supports several flavors of nullability annotations,
+##### Type qualifier nicknames (since Kotlin 1.1.60)
+
+If an annotation type is annotated with both
+[`@TypeQualifierNickname`](https://aalmiray.github.io/jsr-305/apidocs/javax/annotation/meta/TypeQualifierNickname.html) 
+and JSR-305 `@Nonnull` (or its another nickname, such as `@CheckForNull`), then the annotation type is itself used for 
+retrieving precise nullability and has the same meaning as that nullability annotation:
+
+``` java
+@TypeQualifierNickname
+@Nonnull(when = When.ALWAYS)
+@Retention(RetentionPolicy.RUNTIME)
+public @interface MyNonnull {
+}
+
+@TypeQualifierNickname
+@CheckForNull // a nickname to another type qualifier nickname
+@Retention(RetentionPolicy.RUNTIME)
+public @interface MyNullable {
+}
+
+interface A {
+    @MyNullable
+    String foo(@MyNonnull String x);
+}
+```
+
+`A::foo` in this example is available in Kotlin as `fun foo(x: String): String?`
+
+##### Type qualifier defaults (since Kotlin 1.1.60)
+
+[`@TypeQualifierDefault`](https://aalmiray.github.io/jsr-305/apidocs/javax/annotation/meta/TypeQualifierDefault.html) 
+allows introducing annotations that, when being applied, define the default nullability within the scope of the annotated 
+element.
+
+Such annotation type should itself be annotated with both `@Nonnull` (or its nickname) and `@TypeQualifierDefault(...)` 
+with one or more `ElementType` values:
+* `ElementType.METHOD` for return types of methods;
+* `ElementType.PARAMETER` for value parameters;
+* `ElementType.FIELD` for fields; and
+* `ElementType.TYPE_USE` for any type including type arguments, upper bounds of type parameters and wildcard types.
+
+The default nullability is used when a type itself is not annotated by a nullability annotation, and the default is
+determined by the innermost enclosing element annotated with a type qualifier default annotation with the 
+`ElementType` corresponding to the type usage.
+
+```java
+@Nonnull
+@TypeQualifierDefault({ElementType.METHOD, ElementType.PARAMETER})
+public @interface NonNullApi {
+}
+
+@Nonnull(when = When.MAYBE)
+@TypeQualifierDefault({ElementType.METHOD, ElementType.PARAMETER})
+public @interface NullableApi {
+}
+
+@NullableApi
+interface A {
+    String foo(String x); // fun foo(x: String?): String?
+ 
+    @NotNullApi // overriding default from the interface
+    String bar(String x, @Nullable String y); // fun bar(x: String, y: String?): String 
+    
+    // The type of `x` parameter remains platform because there's explicit UNKNOWN-marked
+    // nullability annotation
+    String baz(@Nonnull(when = When.UNKNOWN) String x); // fun baz(x: String!): String?
+}
+```
+
+Package-level default nullability is also supported:
+
+```java
+// FILE: test/package-info.java
+@NonNullApi // declaring all types in package 'test' as non-nullable by default
+package test;
+```
+
+##### `@UnderMigration` annotation (since Kotlin 1.1.60)
+
+The `@UnderMigration` annotation (from `kotlin.annotations.jvm`) can be used to define a migration status
```

Comment:
Note, that these annotations belong to a separate library artifact called `kotlin-annotations-jvm` (in the end it should available in maven/gradle along with other artifacts)

Reply:
Also I'd explain/give an example on how it can be used by a library maintainer